Fix `DEBUGLOG_WITH_STREAM_TYPE_AND_FILE` broken in #150750

ingomueller-net opened the pull request:
This PR fixes the `DEBUGLOG_WITH_STREAM_TYPE_AND_FILE` macro that got broken in #150750. That PR introduces a more sophisitaced version of that macro and refactored some code in that process, making the `getShortFileName` a free function instead of a class member function, but did not adapt this macro to the refactored code.

ingomueller-net committed:
Fix `DEBUGLOG_WITH_STREAM_TYPE_AND_FILE` broken in #150750

This PR fixes the `DEBUGLOG_WITH_STREAM_TYPE_AND_FILE` macro that got
broken in #150750. That PR introduces a more sophisitaced version of
that macro and refactored some code in that process, making the
`getShortFileName` a free function instead of a class member function,
but did not adapt this macro to the refactored code.

Signed-off-by: Ingo Müller <ingomueller@google.com>

joker-eph reviewed (approved):
Thanks!
We don't exercise this code path upstream right now.

What about adding this diff to your PR:

```
diff --git a/llvm/unittests/Support/DebugLogTest.cpp b/llvm/unittests/Support/DebugLogTest.cpp
index 8356ae856be2..3c82376734b9 100644
--- a/llvm/unittests/Support/DebugLogTest.cpp
+++ b/llvm/unittests/Support/DebugLogTest.cpp
@@ -6,6 +6,11 @@
 //
 //===----------------------------------------------------------------------===//
 
+// This macro is defined in the LLVM build system, but we undefine it here
+// so that we test at least once in-tree the case where __SHORT_FILE__ is not
+// defined.
+#undef __SHORT_FILE__
+
 #include "llvm/Support/DebugLog.h"
 #include "llvm/ADT/Sequence.h"
 #include "llvm/Support/raw_ostream.h"
```

ingomueller-net commented:
Sounds good! Will update the PR shortly.

ingomueller-net committed:
make previously broken macro being tested

As suggested by @joker-ehp, this commit undefines `__SHORTFILE__` in
`DebugLogTest.cpp` order to make the previously broken variant of
`DEBUGLOG_WITH_STREAM_AND_TYPE` being tested at least in this one test,
as it is otherwise never used in CI.

Signed-off-by: Ingo Müller <ingomueller@google.com>

joker-eph reviewed (approved):
Thanks for the fix!
